gnosis/src/main/scala/edu/uci/ics/cloudberry/gnosis/NewYorkGnosis.scala

```diff
@@ -0,0 +1,66 @@
+package edu.uci.ics.cloudberry.gnosis
+
+import java.io.{File, FilenameFilter}
+
+import com.vividsolutions.jts.geom.{Coordinate, Envelope, Geometry}
+import play.api.libs.json.{JsObject, Json, Writes}
+
+class NewYorkGnosis(levelGeoPathMap: Map[TypeLevel, File]) {
```

Comment:
Similar as previous `NYGeoIndex`, could you extract the common requirement from the both `gnosis` as some `Abstract` class, and let both concrete class extend on it?